### DISCUSSION TITLE
rSPARQL-anything package idea

### DISCUSSION THREAD

**Discussion by antaldaniel** [SPARQL-Anything/sparql.anything#406] [Ideas]
I am considering to create and maintain an R binding similar to pysparql-anything 0.8.1.1, using rJava. R is not a general purpose language, it is mainly used for statistics and data analysis, but using SPARQL and RDF in statistical data production and interchange is very common and I think that this would open a large potential user base.

The rJava package allows running Java code [directly from R](https://jozef.io/r901-primer-java-from-r-1/#calling-java-from-r-directly), but I assume that we would rather [start from](https://jozef.io/r902-primer-java-from-r-2/) the .jar file.

Here is a dummy R package, [helloJavaWorld](https://cran.r-project.org/web/packages/helloJavaWorld/index.html),  that demonstrates how to interface to a jar file that resides inside an R package. In my opinion, what would be required to create an R package that places the jar file into an R package with wrapper functions in R and a verbose documentation for R users on installation and trying out an example (the one that you provide for the command line tool.)

**Comment by luigi-asprino** [comment]
Hi Daniel,

It's wonderful to hear that you're considering creating an R binding for SPARQLAnything using rJava! Your vision for expanding the potential user base within the R community for such a tool is certainly compelling. I'm not familiar with R nor rJava, but if you have any specific questions on SPARQLAnything, feel free to reach out.

**Comment by enridaga** [comment]
Fantastic proposition, Daniel, I've invited to the SPARQL-Anything organisation so you can start setting up things in this space. Let us know how we can help :)

**Comment by antaldaniel** [comment]
I just forwarded my ideas to the rOpenGov community with whom I work with (I have 6 released packages on CRAN that handles extension libraries to R) The coding work seems very straighforward, I want to discuss maintainance and documentation; for example, it is clear that each time you release a new Java version we have to release the matching R package release. It is not a big deal, perhaps an hour if there are no breaking changes, with documentation included, but I want to ensure that if we do this than the R package will be as well documented and maintained as the Python or the Java original. I will include my contributors and get back shortly with questions.  If we go ahead, for example, I will use the same pre-existing files for unit testing that you use in Python or other side-projects.
